editor/src/messages/tool/tool_messages/path_tool.rs

```diff
@@ -72,13 +107,49 @@ impl ToolMetadata for PathTool {
 
 impl LayoutHolder for PathTool {
 	fn layout(&self) -> Layout {
-		Layout::WidgetLayout(WidgetLayout::default())
+		if let Some(SingleSelectedPoint { coordinates: DVec2 { x, y }, .. }) = self.tool_data.single_selected_point {
+			let x_loc = NumberInput::new(Some(x))
+				.unit(" px")
+				.label("x")
+				.min(-((1u64 << std::f64::MANTISSA_DIGITS) as f64))
+				.max((1u64 << std::f64::MANTISSA_DIGITS) as f64)
+				.on_update(move |number_input: &NumberInput| {
+					let new_x = number_input.value.unwrap_or(x);
+					PathToolMessage::SelectedPointXChanged { new_x }.into()
+				})
+				.widget_holder();
+
+			let y_loc = NumberInput::new(Some(y))
+				.unit(" px")
+				.label("y")
+				.min(-((1u64 << std::f64::MANTISSA_DIGITS) as f64))
+				.max((1u64 << std::f64::MANTISSA_DIGITS) as f64)
+				.on_update(move |number_input: &NumberInput| {
+					let new_y = number_input.value.unwrap_or(y);
+					PathToolMessage::SelectedPointYChanged { new_y }.into()
+				})
+				.widget_holder();
+
+			let seperator = Separator::new(SeparatorType::Unrelated).widget_holder();
+
+			Layout::WidgetLayout(WidgetLayout::new(vec![LayoutGroup::Row {
+				widgets: vec![x_loc, seperator, y_loc],
+			}]))
+		} else {
+			Layout::WidgetLayout(WidgetLayout::default())
+		}
 	}
 }
 
 impl<'a> MessageHandler<ToolMessage, &mut ToolActionHandlerData<'a>> for PathTool {
 	fn process_message(&mut self, message: ToolMessage, responses: &mut VecDeque<Message>, tool_data: &mut ToolActionHandlerData<'a>) {
 		self.fsm_state.process_event(message, &mut self.tool_data, tool_data, &(), responses, true);
+
+		let new_point = get_single_selected_point(&tool_data.document.document_legacy, tool_data.shape_editor);
```

Comment:
Is it idiomatic in this code base to run code on every message? most messages can change the selected point so it seemed appropriate here.

Reply:
Good idea to ask! While not terrible, I think it might be slightly more idiomatic to create another message called something like "selected points count changed" and make sure that any message handler which modifies the count calls that. Alternatively, that could be an event, which is a special type of message that allows subscriptions by users who need that information (see `editor/src/messages/broadcast/broadcast_event.rs`).

Reply:
I'm going to go with the first option. To do the second properly I feel like we would have to make every tool broadcast it's updates to the selected point set, or diff the set repeatedly, which seems out of scope for this PR. Let me add a new message and move the point update code to the relevant match case.

Reply:
Perfect, and yeah, I was mostly mentioning the event concept for your future awareness. I agree using a regular message is likely more appropriate here.

Reply:
thanks!

Reply:
I added the message, the `send_layout` call still has to happen outside the main match body because the `LayoutHolder` implementation is on `PathTool`.